apps/frontend/src/pages/Profile.tsx

```diff
@@ -0,0 +1,264 @@
+        };
+
+        reader.readAsDataURL(file); // This will trigger the onload function above
+      }
+    } else {
+      alert("No image selected.");
+    }
+
+    setIsImageUploaded(true);
+  };
+
+  const handleInputChange = (
+    e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+    field: string,
+  ) => {
+    setProfile({ ...profile, [field]: e.target.value });
+    setEditMode(true);
+  };
+
+  const handleSaveChanges = async () => {
+    let successImages: boolean | ImageURLObject = false;
+    if (isImageUploaded) {
+      try {
+        const listingImageBinary = imageURL.split(",")[1];
+        successImages = await asyncUploadSingleImage(listingImageBinary);
+      } catch (error) {
+        console.error("Error processing image:", error);
+      }
+    } else {
+      successImages = { url: profile.profile_picture };
+    }
+    if (successImages) {
+      profile.profile_picture = successImages.url;
+      try {
+        await _axios_instance.patch("/user", profile);
+        setOriginalProfile(profile);
+        console.log("Profile updated successfully");
+      } catch (error) {
+        console.error("Error saving changes:", error);
+        alert("Error saving changes. Please try again.");
+      }
+    } else {
+      alert("Error uploading picture. Please try again.");
+    }
+
+    setEditMode(false);
+    setIsImageUploaded(false);
+  };
+
+  const handleCancelChanges = () => {
+    setProfile(originalProfile);
+    setEditMode(false);
+  };
+
+  // TODO:
+  // Uncomment once Scott's PR is merged: https://github.com/UVicMartletplace/martletplace/pull/183/files
+  // Disable all text fields if !isCurrentUser
+  // Hide Save, Cancel, and Upload Profile buttons if !isCurrentUser
+  // Change "My Profile" to "<Name>'s Profile" if !isCurrentUser
+  // Check if id is the current user's id:
+  // const { user } = useUser();
+  // const [isCurrentUser, setIsCurrentUser] = useState(false);
+  // if (id === user.id) {
+  //   setIsCurrentUser(true);
+  // }
+
+  return (
+    <Box
+      sx={{
+        display: "flex",
+        flexDirection: "column",
+        justifyContent: "center",
+        alignItems: "center",
+        height: "100vh",
+      }}
+    >
+      {isDesktop && !isScreenSmall && (
+        <AccountSidebar selectedItem="My Profile" />
+      )}
+      <Typography variant={"h2"}>My Profile</Typography>
+      <Avatar
+        src={imageURL}
+        alt="Profile Picture"
+        sx={{ width: 250, height: 250, mt: 2, mb: 2 }}
+        id="profile_picture"
+      />
+      <Button
+        variant="contained"
+        component="label"
+        sx={classes.uploadPfp}
+        id="upload_button"
+      >
+        Upload Picture
+        <input type="file" hidden onChange={handleImageUpload} />
+      </Button>
+      <Box component="form" sx={{ width: "80%", maxWidth: "400px" }}>
+        <TextField
+          label="Username"
+          variant="outlined"
+          sx={{ mt: 2, width: "100%" }}
```

Comment:
Typically it would be preferred to do the margin here on the outer box instead of the TextField, but that is the nittiest of nitpicks

Reply:
do you mean the `margin="normal"` should be added to the box?

I might be misunderstanding, because that didn't work for me